lib/disk_based_memory.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class DiskBasedMemory
+
+
+  def state
+    running_dir = File.dirname(__FILE__)
```

Comment:
I believe you can do `__dir__` (lowercase) to get the directory

Reply:
We tried.. and it breaks every test. So we going to leave it for now as it's too expensive to implement currently.